src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java

```diff
@@ -329,6 +330,52 @@ public void withUpsert() {
           .body("data.documents[0]", jsonEquals(expected));
     }
 
+    @Test
+    public void withUpsertNoId() {
+      String json =
+          """
+        {
+          "findOneAndReplace": {
+            "filter" : {"username" : "username2"},
+            "replacement" : {"username": "username2", "status" : true },
+            "options" : {"returnDocument" : "after", "upsert" : true}
+          }
+        }
+        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.document._id", is(notNullValue()))
+          .body("status.matchedCount", is(0))
+          .body("status.modifiedCount", is(0))
+          .body("status.upsertedId", is(notNullValue()))
+          .body("errors", is(nullValue()));
+
+      // assert state after update
+      json =
+          """
+        {
+          "find": {
+            "filter" : {"username" : "username2"}
+          }
+        }
+        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents[0]._id", is(notNullValue()));
```

Comment:
Might be also good to check that the document's `_id` is a string

Reply:
Will do